compiler/AST/primitive.cpp

```diff
@@ -281,9 +281,11 @@ static Type*
 returnInfoEndCount(CallExpr* call) {
   static Type* endCountType = NULL;
   if (endCountType == NULL) {
-    forv_Vec(TypeSymbol, ts, gTypeSymbols) {
-      if (!strcmp(ts->name, "_EndCount")) {
-        endCountType = ts->type;
+    // Look for the type var `_remoteEndCountType` in ChapelBase.
+    forv_Vec(VarSymbol, var, gVarSymbols) {
+      const char* searchStr = "_remoteEndCountType";
+      if (strcmp(var->cname, searchStr) == 0) {
+        endCountType = var->type;
```

Comment:
Your PR that does this is #2447, right? I'm OK with it being a well-known type, as it really should only be manipulated by calls inserted by the compiler. I'm less sure about splitting it into two (or more) types. For the most part I think the various uses of _EndCount will want to share most of their code for the manipulating functions, and I'm wondering if it would be a lot of code duplication to have several different _EndCount types.

Anyways, I'm in favor of the compiler being able to know more about _EndCount in general.